.github/workflows/preview-docs.yml

```diff
@@ -0,0 +1,162 @@
+name: Preview Documentation (Azure Blob Storage)
+
+on:
+    pull_request:
+        types: [opened, synchronize, reopened, closed]
+        branches:
+            - '**'
+
+permissions:
+    contents: read
+    pull-requests: write
+
+jobs:
+    build_and_deploy_job:
+        if: github.event_name == 'pull_request' && github.event.action != 'closed'
+        runs-on: ubuntu-latest
+        name: Build and Deploy Job
+        outputs:
+            doc_url: ${{ steps.deploy.outputs.docs_url }}
+        steps:
+            ## --- SETUP --- ##
+            - name: Check out code
+              uses: actions/checkout@v4
+              with:
+                  fetch-depth: 0
+
+            - name: Use Node LTS version
+              uses: actions/setup-node@v4
+              with:
+                  node-version: 20
+                  cache: yarn
+
+            - name: Enable Corepack
+              run: corepack enable
+
+            - name: Generate PR hash
+              id: pr_hash
+              run: |
+                  # Use just PR number so each commit overwrites the previous deployment
+                  pr_hash="pr-${{ github.event.pull_request.number }}"
+                  echo "hash=${pr_hash}" >> $GITHUB_OUTPUT
+                  echo "Generated PR hash: ${pr_hash}"
+
+            ## --- YARN CACHE --- ##
+            - name: Check for cached dependencies
+              continue-on-error: true
+              id: cache-dependencies
+              uses: actions/cache@v4
+              with:
+                  path: |
+                      .cache/yarn
+                      node_modules
+                  key: ubuntu-latest-node20-${{ hashFiles('yarn.lock') }}
+
+            ## --- INSTALL & BUILD --- ##
+            - name: Install dependencies
+              shell: bash
+              run: yarn install --immutable
+
+            - name: Build
+              run: yarn build
+
+            - name: Generate Custom Elements Manifest
+              run: yarn docs:analyze
+
+            - name: Move CEM to Storybook directory
+              run: cp projects/documentation/custom-elements.json storybook/
+
+            - name: Build documentation with path prefix
+              env:
+                  SWC_DIR: ${{ steps.pr_hash.outputs.hash }}/docs
+              run: |
+                  cd projects/documentation
+                  yarn build
+
+            - name: Build Storybook
+              run: yarn storybook:build
+
+            ## --- DEPLOY TO AZURE BLOB STORAGE --- ##
+            - name: Install AzCopy
+              run: |
+                  # Download and install AzCopy
+                  wget -O azcopy.tar.gz https://github.com/Azure/azure-storage-azcopy/releases/download/v10.29.1/azcopy_linux_amd64_10.29.1.tar.gz
```

Comment:
We're hard-coding the AzCopy version here, but we have access to the [latest releases](https://api.github.com/repos/Azure/azure-storage-azcopy/releases/latest), could we curl it and grep the platform we want?

Reply:
I only suggested this just seeing some cons which might introduced variability into our builds due to potential breakage if a release is  pulled or changes format